api/views.py

```diff
@@ -0,0 +1,176 @@
+import datetime
+
+from django.contrib.auth.models import User, Group
+from rest_framework import viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from api.serializers import UserSerializer, GroupSerializer
+from .models import Employee, Income, Outcome, Event
+from .serializers import EmployeeSerializer, EventMiniSerializer, IncomeSerializer, OutcomeSerializer, EventSerializer
+from datetime import timedelta, datetime
+
+
+class UserViewSet(viewsets.ModelViewSet):
+    queryset = User.objects.all().order_by('-date_joined')
+    serializer_class = UserSerializer
+
+
+class GroupViewSet(viewsets.ModelViewSet):
+    queryset = Group.objects.all()
+    serializer_class = GroupSerializer
+
+
+class EmployeeViewSet(viewsets.ModelViewSet):
+    queryset = Employee.objects.all()
+    serializer_class = EmployeeSerializer
+
+    def list(self, request, *args, **kwargs):
+        number_of_employees = request.GET.get('number_of_employees', None)
+
+        if number_of_employees is not None:
+            employees = Employee.objects.order_by('-salary')[:number_of_employees]
+        else:
+            employees = Employee.objects.order_by('-salary')
+
+        serializer = self.get_serializer(employees, many=True)
+        return Response(serializer.data)
+
+    '''Fires employee based on given pesel number'''
+
+    @action(detail=False, methods=['get'])
+    def fire(self, request, *args, **kwargs):
+        pesel = request.GET.get('pesel', None)
+        employee = Employee.objects.filter(pesel=pesel)
+
+        if pesel is not None:
+            if not employee:
+                return Response("There is no employee with pesel: " + pesel)
+            else:
+                first_name = employee.get().first_name
+                last_name = employee.get().last_name
+                employee.delete()
+                return Response("Employee " + first_name + " " + last_name + " fired.")
+
+    '''promotes\demotes employe based on given pesel number'''
+
+    @action(detail=False, methods=['get'])
+    def change_position(self, request, *args, **kwargs):
+        pesel = request.GET.get('pesel', None)
```

Comment:
Same as above, use DRF built-in params handling for PESEL (it works here as `<id>` part in URL in fact and should be handled in the url: `/employees/<id>/change_position`.